pkgs/by-name/co/colmap/package.nix

```diff
@@ -22,71 +25,101 @@
   cudaSupport ? config.cudaSupport,
   cudaCapabilities ? cudaPackages.flags.cudaCapabilities,
   cudaPackages,
+  faiss,
+  llvmPackages,
+  gtest,
 }:
 
 assert cudaSupport -> cudaPackages != { };
 
 let
-  boost_static = boost.override { enableStatic = true; };
   stdenv' = if cudaSupport then cudaPackages.backendStdenv else stdenv;
 
+  # These deps are also needed by pycolmap.
+  pythonDeps = [
+    boost
+    eigen
+    ceres-solver
+    openimageio
+    glog
+    libGLU
+    glew
+    cgal
+    poselib
+    faiss
+    sqlite
+    gmp
+    mpfr
+    lz4
+    qt5.qtbase
+  ]
+  ++ lib.optionals cudaSupport [
+    cudatoolkit
+    cudaPackages.cuda_cudart.static
+  ]
+  ++ lib.optional stdenv'.cc.isClang llvmPackages.openmp;
+
   # TODO: migrate to redist packages
   inherit (cudaPackages) cudatoolkit;
 in
 stdenv'.mkDerivation rec {
-  version = "3.9.1";
+  version = "unstable-2025-08-21";
   pname = "colmap";
   src = fetchFromGitHub {
     owner = "colmap";
     repo = "colmap";
-    rev = version;
-    hash = "sha256-Xb4JOttCMERwPYs5DyGKHw+f9Wik1/rdJQKbgVuygH8=";
+    rev = "d478af0ce448251b44e6a4525f8bf640f6cdc9e3";
+    hash = "sha256-UOPm9p5i0+cMmGQYixHDwtxtdM1taHD0QvVyCJPTico=";
   };
 
-  cmakeFlags = lib.optionals cudaSupport [
-    (lib.cmakeBool "CUDA_ENABLED" true)
+  # TODO: remove this when OpenImageIO support is in a release
+  patches = [ ./openimageio.patch ];
```

Comment:
3500 line patch. Does that have to be part of nixpkgs? Otherwise we should `fetchpatch` that if possible.

I also couldn't find how this patch is obtained. Is it some subset of https://github.com/colmap/colmap/pull/3459/commits ?

Reply:
The new version of the PR shows these instructions. I verified that they those instructions produce the patch exactly.